README.md

```diff
@@ -1,241 +1,296 @@
+[Give permissions to submit GPU workloads](#give-permissions-to-submit-gpu-workloads).
 
-Unless using the flag `--rm` when starting a container, you can restart it after it was exited, with the command:
+Unless using the flag `--rm` when starting a container, you can restart it after it was
+exited, with the command:
 
-```
+```bash
 sudo docker start -a -i my_container_name
 ```
 
-Once inside the container, you can check that the environment works: the command
-
-```
-python -c "import dpctl; print(dpctl.get_devices())"
-```
+Once you have loaded into the container, follow those instructions to install the
+`wip-engines` branch of scikit-learn:
 
-will introspect the available hardware, and should display working `opencl` cpu and gpu devices, and `level_zero` gpu devices.
-
-### Step 2: install the `wip-engines` branch of scikit-learn
-
-Once you have loaded into a `numba_dpex` development environment, following one of the two previous guides, follow those instructions:
-
-```
+```bash
 git clone https://github.com/ogrisel/scikit-learn -b wip-engines
 cd scikit-learn
-# NB: recommended commit is subject to change without notice
 git checkout fdaf97b5b90e18fc63483de9455970123208c9bb
-pip install -e . --no-build-isolation
+pip install -e .
 cd ..
 ```
 
-### Step 3: install this plugin
+### Step 2: Check the installation of the environment was successfull
+
+Once the environment you just installed with one of those two methods is activated,
+you can inspect the available hardware:
 
+```bash
+python -c "import dpctl; print(dpctl.get_devices())"
 ```
+
+this should print a list of available devices, including `cpu` and `gpu` devices, once
+for each available backends (`opencl`, `level_zero`,...).
+
+### Step 3: install this plugin
+
+FIXME: currently, non-editable mode installation does not work.
+
+When loaded into your `numba_dpex` + `scikit-learn` environment from previous steps,
+run:
+
+```bash
 git clone https://github.com/soda-inria/sklearn-numba-dpex
 cd sklearn-numba-dpex
-pip install -e . --no-build-isolation
+pip install -e .
 ```
 
 ## Intended usage
 
 See the `sklearn_numba_dpex/kmeans/tests` folder for example usage.
 
-TODO: write some doc here instead.
+🚧 TODO: write some examples here instead.
 
 ### Running the tests
 
-To run the tests run the following:
+To run the tests run the following from the root of the `sklearn_numba_dpex` repository:
 
-```
-pytest sklearn_numba_dpex/
+```bash
+pytest sklearn_numba_dpex
 ```
 
-To run the `scikit-learn` tests with the `sklearn_numba_dpex` engine you can run the following:
+To run the `scikit-learn` tests with the `sklearn_numba_dpex` engine you can run the
+following:
 
-```
-pytest --sklearn-engine-provider sklearn_numba_dpex --pyargs sklearn.cluster.tests.test_k_means
+```bash
+SKLEARN_NUMBA_DPEX_TESTING_MODE=1 pytest --sklearn-engine-provider sklearn_numba_dpex --pyargs sklearn.cluster.tests.test_k_means
```

Comment:
Unrelated comment: I think we should be able to define `conftest.py` not to specify `SKLEARN_NUMBA_DPEX_TESTING_MODE`.

Reply:
Since those are `scikit-learn` tests such a `conftest` should be located in `scikit-learn` I think ? and that would not be possible because this is a parameter specific to `sklearn-numba-dpex`

Reply:
OK let's leave it as is for now.